logstage/logstage-config/src/main/scala/com/github/pshirshov/izumi/logstage/config/codecs/LogSinkCodec.scala

```diff
@@ -0,0 +1,59 @@
+package com.github.pshirshov.izumi.logstage.config.codecs
+
+import com.github.pshirshov.izumi.fundamentals.reflection.SafeType0
+import com.github.pshirshov.izumi.fundamentals.typesafe.config.{ConfigReader, RuntimeConfigReader, RuntimeConfigReaderCodecs, RuntimeConfigReaderDefaultImpl}
+import com.github.pshirshov.izumi.logstage.api.logger.LogSink
+import com.github.pshirshov.izumi.logstage.api.rendering.RenderingPolicy
+import com.github.pshirshov.izumi.logstage.config.codecs.LogSinkCodec.{LogSinkMapper, _}
+import com.typesafe.config.{Config, ConfigFactory, ConfigObject, ConfigValue}
+
+import scala.reflect.runtime.universe
+import scala.util.Try
+
+class LogSinkCodec(policyMappers: Set[LogSinkMapper[_ <: LogSink, _]]) extends ConfigReader[LogSink] {
```

Comment:
`_ <: LogSink` should be virtually the same as `LogSink` given that LogSinkMapper has `+` on first parameter

Reply:
Actually parameter from LogSinkMapper can be removed - users can just covariantly override `apply` and `instantiate` to get more specific type, or moved to type member